File: Python programs/Control Flow/count_digits.py
# Program to count the number of digits

# User enters number
number = int( input('Enter an integer: ') )

# Define an indexing variable
count = 0

# Divide number by 10 to cut off the last digit
# Advance count by 1 and continue until number is 0
while number != 0:
    number = int(number / 10)
    count += 1

# Print the number of digits ( number of interations )
print('Number of digits: ', count )

File: C programs/Count_Digits.c
// Program to count the number of digits in an integer number
#include <stdio.h>

int main( void ) {
    int number, num_save, count = 0;

    // User enters integer to be checked
    printf("Please enter an integer: ");
    scanf(" %d", &number);

    // Save original number to print later
    num_save = number;

    // Loop to count digits
    do {
        number /= 10;
        count++;
    } while ( number != 0 );

    // Print number of digits
    printf("Number of digits in %d is %d", num_save, count);

    // Exit program
    return 0;
}
